criteria.go

```diff
@@ -156,6 +156,33 @@ func (f *field) In(values ...interface{}) *Criteria {
 	}
 }
 
+func (f *field) Contains(elems ...interface{}) *Criteria {
+	return &Criteria{
+		p: func(doc *Document) bool {
+			docValue := doc.Get(f.name)
+			if docValue == nil || reflect.TypeOf(docValue).Kind() != reflect.Slice {
+				return false
+			}
+			rv := reflect.ValueOf(docValue)
+			len := rv.Len()
+			set := make(map[interface{}]int)
```

Comment:
Use a `map[interface{}]bool` instead

Reply:
I can add a few more test cases. According to this G4G page [it is possible to use structs as map keys](https://www.geeksforgeeks.org/golang-program-that-uses-structs-as-map-keys/#:~:text=In%20maps%2C%20most%20of%20the,types%20into%20a%20single%20type.).

Reply:
What about `maps` and `slices`?

Reply:
Using quadratic runtime version now